### PR TITLE
Create LoadFlowResultJsonModule

### PR DESCRIPTION
**Pull request by etiennehomer**
Signed-off-by: HOMER Etienne <etiennehomer@gmail.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No.

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
This PR adds the LoadFlowResultJsonModule class for JSON serialization and deserialization.

**What is the current behavior?** *(You can also link to an open issue here)*
The class is absent.

**What is the new behavior (if this is a feature change)?**
The class is created. It's possible to use it for JSON serialization and deserialization.

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
No.

### CODE REVIEW

File: loadflow/loadflow-api/src/main/java/com/powsybl/loadflow/json/LoadFlowResultJsonModule.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (c) 2018, All partners of the iTesla project (http://www.itesla-project.eu/consortium)
```
Comment (floriand-e2r) [inline loadflow/loadflow-api/src/main/java/com/powsybl/loadflow/json/LoadFlowResultJsonModule.java:2]: Please update the copyright year. Is the copyright iTesla or RTE btw?

File: loadflow/loadflow-api/src/test/java/com/powsybl/loadflow/json/LoadFlowResultJsonTest.java [hunk 2]
```diff
@@ -62,4 +67,17 @@ public void handleErrorTest() throws IOException {
...
+    public void loadFlowResultJsonModuleTest() throws IOException {
+        ObjectMapper objectMapper = JsonUtil.createObjectMapper()
+                .registerModule(new LoadFlowResultJsonModule());
+        ObjectWriter writer = objectMapper.writerWithDefaultPrettyPrinter();
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+        writer.writeValue(os, createVersion11());
+        String result = new String(os.toByteArray(), StandardCharsets.UTF_8);
+
+        compareTxt(getClass().getResourceAsStream("/LoadFlowResultVersion11.json"), result);
```
Comment (floriand-e2r) [inline loadflow/loadflow-api/src/test/java/com/powsybl/loadflow/json/LoadFlowResultJsonTest.java:80]: You should also test that you can deserialize the string properly through an ObjectReader.

**Review by floriand-e2r — CHANGES_REQUESTED** [review]

File: loadflow/loadflow-api/src/main/java/com/powsybl/loadflow/json/LoadFlowResultJsonModule.java (hunk 1 above)
Comment (mathbagu) [reply loadflow/loadflow-api/src/main/java/com/powsybl/loadflow/json/LoadFlowResultJsonModule.java:2]: `Copyright (c) 2020, RTE (https://ww.rte-france.com)`

File: loadflow/loadflow-api/src/test/java/com/powsybl/loadflow/json/LoadFlowResultJsonTest.java (hunk 2 above)
Comment (mathbagu) [reply loadflow/loadflow-api/src/test/java/com/powsybl/loadflow/json/LoadFlowResultJsonTest.java:80]: Agree: a good and complete test will do a complete round-trip. I think you should also update the `LoadFlowResultSerializer::write` and `LoadFlowResultDeserializer::read`. In these two methods we create manually the module: if you replace this part by the module you have just created you will cover your new code.

**Review by mathbagu — CHANGES_REQUESTED** [review]

**Review by floriand-e2r — APPROVED** [review]

**Review by mathbagu — APPROVED** [review]

**Pull request merged**